packages/web3-eth-transaction-utils/src/utils/get_transaction_gas_pricing.ts

```diff
@@ -31,7 +37,36 @@ async function getEip1559GasPricing<ReturnFormat extends DataFormat>(
 	web3Context: Web3Context<EthExecutionAPI>,
 	returnFormat: ReturnFormat,
 ): Promise<FormatType<{ maxPriorityFeePerGas?: Numbers; maxFeePerGas?: Numbers }, ReturnFormat>> {
-	const block = await getBlock(web3Context, web3Context.defaultBlock, false, returnFormat);
+	let block;
+	if (isBytes(web3Context.defaultBlock)) {
+		const blockHashFormatted = format(
+			{ eth: 'bytes32' },
+			web3Context.defaultBlock,
+			ETH_DATA_FORMAT,
+		);
+		block = format(
+			blockSchema,
+			await ethRpcMethods.getBlockByHash(
+				web3Context.requestManager,
+				blockHashFormatted,
+				false,
+			),
+			returnFormat,
+		);
+	} else {
+		const blockNumberFormatted = isBlockTag(web3Context.defaultBlock as HexString)
+			? (web3Context.defaultBlock as BlockTag)
+			: format({ eth: 'uint' }, web3Context.defaultBlock as Numbers, ETH_DATA_FORMAT);
+		block = format(
+			blockSchema,
+			await ethRpcMethods.getBlockByNumber(
+				web3Context.requestManager,
+				blockNumberFormatted,
+				false,
+			),
+			returnFormat,
+		);
+	}
```

Comment:
`getBlock` from `web3-eth`'s `rpc_method_wrappers` was originally doing the

```typescript
if (isBytes(web3Context.defaultBlock)) { .... } else { ... }
```
but we're having to do the check here to avoid circular dependency issues since `web3-eth` depends on `web3-eth-transaction-utils` - perhaps this chunk of code could be turned into a util function